build/DownloadCFamilyPlugin/Common.cs

```diff
@@ -0,0 +1,239 @@
+        {
+            var destFile = Path.Combine(destinationDirectory, Path.GetFileName(sourceFilePath)) + ".uncompressed";
+            if (File.Exists(destFile))
+            {
+                LogMessage($"Uncompressed tar file already exists: {destFile}", logger);
+                return destFile;
+            }
+
+            using (Stream xz = new XZStream(File.OpenRead(sourceFilePath)))
+            using (Stream outputStream = new FileStream(destFile, FileMode.CreateNew))
+            {
+                xz.CopyTo(outputStream);
+            }
+            return destFile;
+        }
+
+        public static void UncompressAndUnzipTgz(string tarFilePath, string destinationFolder, TaskLoggingHelper logger)
+        {
+            // The txz file is compressed using XZ compression and zipped using the tar format.
+            // There is no built-in framework support for these format so we are using two
+            // open source libraries, both licensed under the MIT license.
+            EnsureWorkingDirectoryExist(destinationFolder, logger);
+
+            var uncompresssedFile = DecompressGZipFile(tarFilePath, destinationFolder, logger);
+            ExtractTarToDirectory(uncompresssedFile, destinationFolder, logger);
+        }
+
+        public static string DecompressGZipFile(string sourceFilePath, string destinationDirectory, TaskLoggingHelper logger)
+        {
+            var destFile = Path.Combine(destinationDirectory, Path.GetFileName(sourceFilePath)) + ".uncompressed";
+            if (File.Exists(destFile))
+            {
+                LogMessage($"Uncompressed tar file already exists: {destFile}", logger);
+                return destFile;
+            }
+
+            using (Stream xz = new GZipStream(File.OpenRead(sourceFilePath), CompressionMode.Decompress))
+            using (Stream outputStream = new FileStream(destFile, FileMode.CreateNew))
+            {
+                xz.CopyTo(outputStream);
+            }
+            return destFile;
+        }
+
+        public static void ExtractTarToDirectory(string sourceFilePath, string destinationDirectory, TaskLoggingHelper logger)
+        {
+            if (Directory.GetDirectories(destinationDirectory).Length > 0)
+            {
+                // Unzipping the jar is slow so skip if possible
+                LogMessage($"Skipping unzipping the tar because the destination folder already contains a sub-folder.", logger);
+                return;
+            }
+
+            using (var outputStream = new FileStream(sourceFilePath, FileMode.Open))
+            {
+                ICSharpCode.SharpZipLib.Tar.TarArchive tarArchive = ICSharpCode.SharpZipLib.Tar.TarArchive.CreateInputTarArchive(outputStream);
+                tarArchive.ExtractContents(destinationDirectory);
+            }
+        }
+
+        public static List<string> FindSingleFiles(string searchRoot, IEnumerable<string> patterns, TaskLoggingHelper logger)
```

Comment:
I split the old "FindFiles" method into this method and the next one so then can be reused more easily for the TypeScript plugin.